accounts/views.py

```diff
@@ -98,10 +100,177 @@ def signup(request):
         form = CustomUserCreationForm()
         return render(request, "accounts/login.html", {"form": form})
 
+
 def settings_view(request):
     return render(request, "accounts/profile_settings.html")
 
+
 def onboarding(request):
     # if(request.session.get("onboarding")):
     return render(request, "accounts/onboarding.html")
 
+
+# --- Friend handling begins ---
+
+
+def send_friend_request(request):
+    """
+    This function handles when a user wants to send a friend request to another user.
+    """
+    # check request is POST and has username field
+    if request.method != "POST":
+        return redirect("landing_page:index")
+    try:
+        data = json.loads(request.body)
+        request_username = data["username"]
+    except (json.JSONDecodeError, KeyError):
+        return HttpResponse("Invalid JSON", status=400)
+    print("request_username: ", request_username)
+    # check user is logged in and User exists
+    if not request.user.is_authenticated:
+        return redirect("accounts:login")
+    try:
+        receiver_user = User.objects.get(username=request_username)
+    except User.DoesNotExist:
+        return HttpResponse("User not found.", status=404)
+    receiver_id = receiver_user.pk
+    # check if user is trying to send a friend request to themselves
```

Comment:
i get that all of these are returning so the need to make them 'if / elif' isn't necessary, but it would make sense in a "here are all the cases we are checking, we're going to return an error if any are true"

Reply:
also i feel like the message in the httpresponse is self-explanatory - i don't know if the comments are that helpful